### PR TITLE
blockdev_commit_cor:add support for cor filter

### PR DESCRIPTION
**Pull request by aliang123**
Commit with copy-on-read filter on top

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:2014969

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_commit_cor.q35: PASS (40.48 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_commit_cor.q35: PASS (46.81 s)

**Comment by aliang123** [comment]
> @aliang123 From test log, it can be seen the copy-on-read filter is on the base instead of the top of the commit chain.

Update the commit with top-node the snapshot node, which will keep cor filter node always on the top.

File: qemu/tests/cfg/blockdev_commit_cor.cfg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    device_tag = "data"
+    images += " ${device_tag}"
+    force_create_image_data = yes
+    remove_image_data = yes
+    image_size_data = 500M
+    image_name_data = data
+    storage_pools = default
+    storage_type_default = "directory"
+    storage_pool = default
+    image_copy_on_read_data = True
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_cor.cfg:16]: yes or on or true, which follows our current design

File: qemu/tests/cfg/blockdev_commit_cor.cfg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    snapshot_tags_data = sn1
+
+    image_name_sn1 = sn1
+    image_format_sn1 = qcow2
+
+    qemu_force_use_drive_expression = no
+    mount_point = "/var/tmp"
```
Comment (zhencliu) [inline qemu/tests/cfg/blockdev_commit_cor.cfg:23]: we can remove it

File: qemu/tests/blockdev_commit_cor.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            if device in info:
+                self.generate_tempfile(info[1], tag)
+
+    def commit_snapshots(self):
+        device_tag = self.params["device_tag"].split()[0]
+        device_params = self.params.object_params(device_tag)
+        snapshot_tags = device_params["snapshot_tags"].split()
+        device = self.get_node_name(device_tag)
+        options = ["top-node"]
+        arguments = self.params.copy_from_keys(options)
```
Comment (zhencliu) [inline qemu/tests/blockdev_commit_cor.py:29]: looks like top-node is set in line 30, so here arguments = {}

Comment (aliang123) [reply qemu/tests/blockdev_commit_cor.py:29]: Yes, arguments={} here. And we set top-node in line 30.

File: qemu/tests/cfg/blockdev_commit_cor.cfg (hunk 2 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_cor.cfg]: Have updated it, thanks.

File: qemu/tests/cfg/blockdev_commit_cor.cfg (hunk 1 above)
Comment (aliang123) [reply qemu/tests/cfg/blockdev_commit_cor.cfg]: Have updated it, thanks zhenchao,

**Comment by zhencliu** [comment]
LGTM

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_commit_cor.q35: PASS (45.38 s)

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.blockdev_commit_cor.q35: PASS (45.27 s)

**Comment by ngu-niny** [comment]
(1/2) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.ppc64le.io-github-autotest-qemu.blockdev_commit_cor: PASS (93.33 s)
(2/2) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_commit_cor: PASS (94.71 s)

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**